Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -240,302 +209,172 @@ function Set-FrontDoorRequiredTLSVersion
-                                                                                    @{N='MinimumTlsVersion';E={$frontDoorEp.MinTLSVersion}},
-                                                                                    @{N='isMinTLSVersionSetOnCustomDomain';E={$frontDoorEp.isMinTLSVersionSetOnCustomDomain}}
-                                    }
-                                }
-                                else
-                                {
-                                    $compliantFrontDoorEPs += $item
-                                }
-                            }
-                            else
-                            {
-                                Write-Host "Skipping this Front End Point..." -ForegroundColor $([Constants]::MessageType.Warning)
-                                Write-Host $([Constants]::SingleDashLine)
-                            }
-                        }
-                    }
-                }
-                catch
-                {
-                    $frontDoorsEPSkipped += $frontDoorResource
-                    Write-Host "Error fetching Front Door configuration: Resource Name: [$($resourceName)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
-                    $logResource = @{}
-                    $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
-                    $logResource.Add("ResourceName",($_.ResourceName))
-                    $logResource.Add("Reason","Encountered error while fetching Front Door configuration")    
-                    $logSkippedResources += $logResource
-                    Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)
-                }
-        }
-    }
-    else
-    {
-        # If filepath is present we are already having the endpoints so just filtering out the non compliant endpoints
-        $FrontDoorResources | ForEach-Object {
-            $frontDoorEndpointResource = $_
+        $validFrontDoorDetails | ForEach-Object {
+            $frontDoorResource = $_
             $resourceGroupName = $_.ResourceGroupName
-            $resourceName = $_.Name
+            $resourceName = $_.ProfileName
             try
             {
-                if($frontDoorEndpointResource.MinimumTlsVersion -ne $requiredMinTLSVersion)
+                Write-Host "Fetching Front Door CDN Profiles Domain configuration: Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+                Write-Host $([Constants]::SingleDashLine)
+                $FrontDoorCustomDomains = @()
+
+                $FrontDoorCustomDomains = Get-AzFrontDoorCdnCustomDomain -ProfileName $resourceName -ResourceGroupName $resourceGroupName -CustomDomainName $_.DomainName
```

Comment:
In the file we are storing row per front door or per custom domain. If it's per custom domain then it's possible when user took snapshot there were say X domains but when user decided to run remediation numbers were different.

Reply:
Hi Tarun, In the else condition I have added the details that error while retrieving the details for that custom domain.